.github/workflows/on-pull-request-opened-and-synchronize.yml

```diff
@@ -258,9 +257,18 @@ jobs:
     permissions:
       contents: read
       pull-requests: write
-    # Ubuntu instead of MacOS because the performance difference is
-    # negligible and Ubuntu includes the AWS CLI.
+
+    # Ubuntu instead of MacOS because the performance difference is negligible and
+    # Ubuntu includes the AWS CLI.
     runs-on: ubuntu-latest
+
+    # Both `github.head_ref` and `github.event.pull_request.head.ref` are empty when
+    # a workflow is run in a merge queue. Take a look at the Deploy Storybook task
+    # below. Because those variables will be empty, letting the task run in a merge
+    # queue means the branch will be deployed to the root of R2 and thus to production.
+    # The point of this job is the Deploy Storybook task. So we just skip the entire job.
+    if: github.event_name != 'merge_group'
```

Comment:
The meat of it. What's still unclear to me is whether a skipped and required check still counts as passing, allowing the branch to be merged into `main`. Skipped status checks _are_ considered neutral. So maybe? 

If not, I'll have to move the `if` down to the Deploy Storybook task below. That way, the overall job is considered successful instead of neutral.

Reply:
https://github.com/CrowdStrike/glide-core/actions/runs/14647517987 👌